doc/nrf/dm_adding_code.rst

```diff
@@ -0,0 +1,147 @@
+As long as you do not need to make any changes to any of the repositories of the |NCS|, you can use the procedures to :ref:`get the source code <dm-wf-get-ncs>` and later :ref:`update it <dm-wf-update-ncs>`, and manage your application separately, inside or outside the top folder of the |NCS|.
+
+If you choose to have your application outside of the folder hierarchy of the |NCS|, the build system will find the location of the SDK through the :makevar:`ZEPHYR_BASE` environment variable, which is set either through a script or manually in an IDE.
+More information about application development and the |NCS| build and configuration system can be found in the :ref:`app_build_system` documentation section.
+
+The drawback with this approach is that any changes you make to the set of |NCS| repositories are not directly trackable using Git, since you do not have any of the |NCS| repositories forked.
+If you are tracking the master branch of the |NCS|, you can instead send the changes you require to the official repositories as Pull Requests, so that they are incorporated into the codebase.
+
+Workflow 3: Application in a fork of `sdk-nrf`_
+===============================================
+
+Forking the `sdk-nrf`_ repository and adding the application to it is another valid option to develop and maintain your application.
+This approach also allows you to fork additional |NCS| repositories and point to those.
+This can be useful if you have to make changes to those repositories beyond adding your own application to the manifest repository.
+
+In order to use this approach, you first need to :ref:`get the source code <dm-wf-get-ncs>`, and then :ref:`fork the sdk-nrf repository <dm-wf-fork>`.
+Once you have your own fork, you can start adding your application to your fork's tree and push it to your own Git server.
+Every time you want to update the revision of the |NCS| that you want to use as a basis for your application, you must follow the :ref:`instructions to update <dm-wf-update-ncs>` on your own fork of ``sdk-nrf``.
+
+If you have changes in additional repositories beyond `sdk-nrf`_ itself, you can point to your own forks of those in the :file:`west.yml` included in your fork.
+
+Workflow 4: Application as the manifest repository
+==================================================
+
+An additional possibility is to take advantage of west to manage your own set of repositories.
+This workflow is particularly beneficial if your application is split among multiple repositories or, just like in the previous workflow, if you want to make changes to one or more |NCS| repositories, since it allows you to define the full set of repositories yourself.
+
+In order to implement this approach you first need to create a manifest repository of your own, which just means a repository that contains a :file:`west.yml` manifest file in its root.
```

Comment:
I would break this section into smaller subsections corresponding to steps, which I counted at least 5 here.

Reply:
This content will get a complete rework later. So, leave as is for now.